170/Add user endpoints

AdamVig opened the pull request:
Closes #170.

AdamVig committed:
Add user endpoints

emroussel commented on controllers/user.js:
```diff
@@ -3,52 +3,66 @@ const endpoint = require('../services/endpoint')
+   *
+   * @apiUse UserResponse
    */
   app.put({name: 'update user', path: 'user/:userID'}, auth.verify, user.update)
```
Does this allow the user to update their password through this endpoint? If yes, it should probably be moved to another endpoint to be more secure.

AdamVig replied:
No, it doesn't. See [this comment](https://github.com/AdamVig/stockpile-api/issues/170#issuecomment-291543580) for an explanation of how I avoided that.

emroussel replied:
Great!

AdamVig committed:
Add roles to all users response for administrator

This will allow the administrator to see each user's role, which is
useful when looking at all users at once.

AdamVig committed:
Add role to get all users endpoint documentation
